### PR TITLE
Implement Destination

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

This PR:

- Implements Destination;

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination/writer/writer.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("get column types: %w", err)
+	}
+	writer.columnTypes = columnTypes
+
+	return writer, nil
+}
+
+// InsertRecord inserts a sdk.Record into a Destination.
+func (w *Writer) InsertRecord(ctx context.Context, record sdk.Record) error {
```
Comment (maha-hajja) [inline destination/writer/writer.go:71]: this method can also delete a record, which makes the name a bit confusing.

**Review by maha-hajja — COMMENTED** [review]
nice work! one mini nitpick and we're good to go :D

**Review by maha-hajja — APPROVED** [review]
perfect!

File: destination/writer/writer.go (hunk 1 above)
Comment (alyakimenko) [reply destination/writer/writer.go]: Yeah, this is a good point. Thank you! I've renamed this method to `Write`.

**Pull request merged**